wave_matching/CMakeLists.txt

```diff
@@ -3,38 +3,56 @@ PROJECT(wave_matching)
 SET(CMAKE_BUILD_TYPE RELEASE)
 
 # PACKAGES
-FIND_PACKAGE(PCL 1.8 REQUIRED)
-FIND_PACKAGE(OpenCV REQUIRED)
 
-# DEFINITIONS
-ADD_DEFINITIONS(${PCL_DEFINITIONS})
+# Define a monolithic imported target for PCL
+FIND_PACKAGE(PCL 1.8 REQUIRED
+    common filters registration kdtree search io visualization)
+ADD_LIBRARY(PCL INTERFACE IMPORTED)
 
-# INCLUDES
-INCLUDE_DIRECTORIES(SYSTEM
-    ${PCL_INCLUDE_DIRS}
-    ${OpenCV_INCLUDE_DIRS})
+# @todo PCL's config file is messy and outdated (overwriting BOOST_LIBRARIES,
+# for example), thus this is messy. (Maybe there is a better solution).
+# We don't use the given PCL_LIBRARIES since it includes "debug" and "optimized"
+# keywords; instead, we specify individual modules. Update if more are needed.
+SET(PCL_LIBRARIES ${PCL_COMMON_LIBRARY}
+    ${PCL_FILTERS_LIBRARY}
+    ${PCL_REGISTRATION_LIBRARY}
+    ${PCL_KDTREE_LIBRARY}
+    ${PCL_SEARCH_LIBRARY}
+    ${PCL_IO_LIBRARY}
+    ${PCL_VISUALIZATION_LIBRARY}
+    ${BOOST_LIBRARIES}
+    ${VTK_LIBRARIES})
 
-# LIBRARY DIRS
+# PCL adds some entries with spaces instead of as a (semicolon-separated) CMake
+# list, which messes up later. Fix them here.
+string(REPLACE " " ";" pcl_definitions "${PCL_DEFINITIONS}")
```

Comment:
@leokoppel you just told me you want to fix this. I'm not sure if this counts as my review anymore.

Reply:
Done, you can review